### PR TITLE
Revisit logic for scheduling an automatic appointment when the user has not scheduled one

### PR DESCRIPTION
**Pull request by msasikanth**
https://www.pivotaltracker.com/story/show/170068992

In the previous PR we are only checking if the blood sugar value is not null. Deepa has added the blood sugar cut off values in the PRD. I have updated the query to include those values.

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/patient/Patient.kt [hunk 1]
```diff
@@ -245,7 +245,9 @@ data class Patient(
             CASE
               WHEN
                 (BP.systolic >= 140 OR BP.diastolic >= 90
-                OR BloodSugar.bloodSugarValue IS NOT NULL
+                OR (BloodSugar.type = "random" AND BloodSugar.value >= 200)
+                OR (BloodSugar.type = "fasting" AND BloodSugar.value >= 300)
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/patient/Patient.kt]: There is a mistake here. The `fasting` and `random` values are switched.
Please refer the PRD and correct these.
Also update the tests accordingly.

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/patient/Patient.kt]: Fixed it 👍

File: app/src/main/java/org/simple/clinic/patient/Patient.kt [hunk 2]
```diff
@@ -245,7 +245,9 @@ data class Patient(
             CASE
               WHEN
                 (BP.systolic >= 140 OR BP.diastolic >= 90
-                OR BloodSugar.bloodSugarValue IS NOT NULL
+                OR (BloodSugar.type = "random" AND BloodSugar.value >= 300)
+                OR (BloodSugar.type = "fasting" AND BloodSugar.value >= 200)
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/patient/Patient.kt]: Can you add test for high `Random`?

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/patient/Patient.kt]: I added `1 Normal Random Blood Sugar`. Should I add a blood sugar which crosses the cut off as well?

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/patient/Patient.kt]: Added a test for high `Random Blood Sugar`

**Pull request merged**